components/scream/scripts/test_all_scream.py

```diff
@@ -298,31 +313,43 @@ def run_all_tests(self):
     def test_all_scream(self):
     ###############################################################################
         success = True
-        # First, create build directories (one per test)
-        for test in self._tests:
-            # Get this test's build dir name and cmake args
-            full_name = self._test_full_names[test]
-            test_dir = "./ctest-build/{}".format(full_name)
+        try:
+            # First, create build directories (one per test)
+            for test in self._tests:
+                # Get this test's build dir name and cmake args
+                full_name = self._test_full_names[test]
+                test_dir = "./ctest-build/{}".format(full_name)
 
-            # Create this test's build dir
-            if os.path.exists(test_dir):
-                shutil.rmtree(test_dir)
+                # Create this test's build dir
+                if os.path.exists(test_dir):
+                    shutil.rmtree(test_dir)
 
-            os.makedirs(test_dir)
+                os.makedirs(test_dir)
+
+            if self._baseline_dir is None:
+                # Second, generate baselines
+                expect(self._baseline_ref is not None, "Missing baseline ref")
+
+                success = self.generate_all_baselines()
+                if not success:
+                    print ("Error(s) occurred during baselines generation phase")
+
+            if self._perform_tests:
+                # Finally, run the tests
+                success &= self.run_all_tests()
+                if not success:
+                    print ("Error(s) occurred during test phase")
+
+        except:
```

Comment:
This is not necessary I don't think. The finally alone should work.

Reply:
makes sense.

Reply:
makes sense.